docs/vectorizer/document-vectorization.md

```diff
@@ -0,0 +1,525 @@
+# Document Vectorization in PGAI
+
+## Introduction
+
+Document vectorization is a critical capability for AI applications that need to work with unstructured data like PDFs, Word documents, or other file formats stored outside PostgreSQL. While RAG (Retrieval Augmented Generation) applications typically require text data, real-world enterprise scenarios often involve documents that:
+
+- Are stored in external systems like S3 or local filesystems
+- Come in various formats (PDF, DOCX, XLSX, EPUB, etc.)
+- Contain complex structures like tables, images, and formatting
+- Change frequently, requiring synchronization between sources and embeddings
+
+PGAI's document vectorization system addresses these challenges through a declarative approach that handles loading, parsing, chunking, and embedding documents with minimal configuration. This architecture:
+
+- Keeps document metadata in PostgreSQL while document content lives in optimized storage systems
+- Automatically processes documents into LLM-friendly formats
+- Maintains synchronization between documents and their embeddings
+- Provides resilient error handling and monitoring
+
+## Setting Up Document Storage
+
+### Document Table Approach
+
+The corner-stone of document management in PGAI is a document metadata table in PostgreSQL, the actual files can either be in the same table as a BYTEA column or in an external storage system like S3. This table serves as the source for vectorization, but you can also add any additional metadata you need for your application.
+
+#### Minimal Document Table
+
+A minimal document source table requires only an identifier and a URI pointing to the document, this can be the same column:
+
+```sql
+CREATE TABLE documents (
+    uri TEXT PRIMARY KEY
+);
+
+-- Example records
+INSERT INTO documents (uri) VALUES 
+    ('s3://my-bucket/documents/product-manual.pdf'),
+    ('s3://my-bucket/documents/api-reference.md'),
+```
+
+#### Extended Document Table
+
+For your system, you can include any additional metadata that you might need to filter or classify documents. To make synchronization easier, you can also include a `created_at` and `updated_at` timestamp, which will trigger re-embedding when they are updated:
+
+```sql
+CREATE TABLE documents (
+    id SERIAL PRIMARY KEY,
+    title TEXT NOT NULL,
+    uri TEXT NOT NULL,
+    content_type TEXT,
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
```

Comment:
Maybe its worth to either mention a trigger will be required to autoupdate this value on each row update, or rather even show how to create the trigger?

Reply:
I mean the whole point here is that you will have to manually update the updated_at from your application, because if just the content of the file changes the postgresdb will otherwise just not know about it. So... I don't think a trigger does anything valuable.